libioc/Jail.py

```diff
@@ -1845,22 +1845,23 @@ def __clear_resource_limits(
 
         yield jailResourceLimitActionEvent.begin()
 
-        config_value = self.config['rlimits']
-        if (config_value is False) or (config_value is None):
+        if self.__resource_limits_enabled is False:
             yield jailResourceLimitActionEvent.skip()
             return
 
         self.logger.verbose("Clearing resource limits")
-        _, _, returncode = libioc.helpers.exec(
-            [
+        stdout, _, returncode = libioc.helpers.exec(
+            " ".join([
                 "/usr/bin/rctl",
                 "-r",
-                f"jail:{self.identifier}"
-            ],
-            ignore_error=True
+                f"jail:{self.identifier}",
+                "2>&1"
+            ]),
+            ignore_error=True,
+            shell=True
         )
 
-        if returncode > 0:
+        if (returncode > 0) and ("No such process" not in stdout):
```

Comment:
wouldn't the string you're looking for be found in `stderr`, which you join (`2>&1`) into `stdout` (which we don't care about at all…)

Reply:
Yes, but stderr handling requires a second thread to observe stderr. It is much easier just to merge the two and look at the combined results.